src/Components/Web/src/Virtualization/ItemsProviderDelegate.cs

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information. 
+
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Components.Virtualization
+{
+    /// <summary>
+    /// A function that provides items to a virtualized source.
+    /// </summary>
+    /// <typeparam name="TItem">The type of the context for each item in the list.</typeparam>
+    /// <param name="startIndex">The start index of the data segment requested.</param>
+    /// <param name="requestedCount">
+    /// The requested number of items to be provided. The actual number of provided items does not need to match
+    /// this value.
+    /// </param>
+    /// <returns>A <see cref="Task"/> whose result is a <see cref="ItemsProviderResult{TItem}"/> upon successful completion.</returns>
+    public delegate Task<ItemsProviderResult<TItem>> ItemsProviderDelegate<TItem>(int startIndex, int requestedCount);
```

Comment:
Does this need to be public?

Reply:
The public parameter `ItemsProvider` in `VirtualizeDeferred` has this type. I created `ItemsProviderDelegate` mostly so the parameters were named and documented, which I thought might be less confusing for users than `Func<int, int ItemsProviderResult<TItem>>`. I suppose one alternative could be something like `Func<ItemsProviderRequest, ItemsProviderResult<TItem>>` or something along those lines as a way to disambiguate `startIndex` and `requestedCount`. Do you have any thoughts about this?

Reply:
`Func<ItemsProviderRequest, ItemsProviderResult<TItem>>` sounds great to me.

The advantage this has over `ItemsProviderDelegate` is that, in the future, we can freely add extra parameters to `ItemsProviderRequest` without it being a breaking change.

Reply:
If you don't like all the nested generics in `Func<ItemsProviderRequest, Task<ItemsProviderResult<TItem>>>` then it would be fine to define `ItemsProviderDelegate` as an equivalent thing to this. I'm just saying it's better to have an `ItemsProviderRequest` type that we can add to in the future rather than fixing it as an `int/int/CancellationToken` triplet forever.